plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/panel/GitPanelPresenter.java

```diff
@@ -0,0 +1,360 @@
+        if (delta.getKind() == ResourceDelta.ADDED) {
+          if ((delta.getFlags() & (MOVED_FROM | MOVED_TO)) != 0) {
+            // project renamed
+            String oldProjectName = delta.getFromPath().segment(0);
+            String newProjectName = delta.getToPath().segment(0);
+            MutableAlteredFiles alteredFiles =
+                new MutableAlteredFiles(
+                    findProjectByName(newProjectName), changes.remove(oldProjectName));
+
+            changes.put(newProjectName, alteredFiles);
+            view.renameRepository(oldProjectName, newProjectName);
+          } else {
+            // project created
+            changes.put(resource.getName(), new MutableAlteredFiles(resource.asProject()));
+            view.addRepository(resource.getName());
+          }
+        } else if (delta.getKind() == ResourceDelta.REMOVED) {
+          // project deleted
+          changes.remove(resource.getName());
+          view.removeRepository(resource.getName());
+        }
+      }
+    } else {
+      // handle deletion of an item
+      if (delta.getKind() == ResourceDelta.REMOVED) {
+        String pathToItem = delta.getResource().getLocation().toString();
+        String projectName = extractProjectName(pathToItem);
+        MutableAlteredFiles alteredFiles = changes.get(projectName);
+        if (alteredFiles == null) {
+          // project doesn't have a git repository
+          return;
+        }
+        String itemRelativePath = removeProjectName(pathToItem);
+
+        if (alteredFiles.getStatusByFilePath(itemRelativePath) != Status.ADDED) {
+          alteredFiles.addFile(itemRelativePath, Status.DELETED);
+        } else {
+          alteredFiles.removeFile(itemRelativePath);
+        }
+
+        view.updateRepositoryChanges(projectName, alteredFiles.getFilesQuantity());
+        if (projectName.equals(selectedProjectName)) {
+          updateChangedFiles(alteredFiles);
+        }
+      }
+    }
+  }
+
+  @Override
+  public void onGitRepositoryInitialized(
+      String endpointId, RepositoryInitializedEventDto repositoryInitializedEventDto) {
+    String projectName = repositoryInitializedEventDto.getProjectName();
+    changes.put(projectName, new MutableAlteredFiles(findProjectByName(projectName)));
+    view.addRepository(projectName);
+  }
+
+  @Override
+  public void onGitRepositoryDeleted(
+      String endpointId, RepositoryDeletedEventDto repositoryDeletedEventDto) {
+    String projectName = repositoryDeletedEventDto.getProjectName();
+    changes.remove(projectName);
+    view.removeRepository(projectName);
+  }
+
+  @Override
+  public void onGitStatusChanged(String endpointId, StatusChangedEventDto dto) {
+    updateRepositoryData(dto.getProjectName());
+  }
+
+  @Override
+  public void onGitCheckout(String endpointId, GitCheckoutEventDto dto) {
+    // this update is needed to correctly handle checkout with force
+    updateRepositoryData(dto.getProjectName());
```

Comment:
I wouldn't like to see a variable named as `dto` in all methods above

Reply:
ok, agree, full name will be better